app/src/main/java/io/gnosis/safe/ui/transactions/details/TransactionDetailsFragment.kt

```diff
@@ -37,12 +50,42 @@ class TransactionDetailsFragment : BaseViewBindingFragment<FragmentTransactionDe
             }
 
             refresh.setOnRefreshListener {
-                //TODO: add on refresh listener
+                viewModel.loadDetails(txId)
             }
         }
 
         viewModel.state.observe(viewLifecycleOwner, Observer { state ->
-
+            when (val viewAction = state.viewAction) {
+                is BaseStateViewModel.ViewAction.Loading -> updateUi(state.txDetails, viewAction.isLoading)
+            }
         })
     }
+
+    override fun onResume() {
+        super.onResume()
+        viewModel.loadDetails(txId)
```

Comment:
if we have pending tx's onResume is good place to update tx details - maybe there are some additional confirmations or other changes. Generally, we should also distinguish btw loading and updating tx data. For creational transaction, there is no sense to have updates or pull to refresh. This can be handled in a separate pr.